Tutorial for spawning a URDF model into Ignition Gazebo

adlarkin opened the pull request:
Signed-off-by: Ashton Larkin <42042756+adlarkin@users.noreply.github.com>

## Summary
I've added a tutorial that explains how to spawn a URDF model in an Ignition Gazebo world.

Currently, the tutorial has only been added to Citadel, but once the tutorial is reviewed/approved, I can update this PR to add the tutorial to other supported Ignition versions as well.

## Test it
Users can test this tutorial by reading through it and following along by running the commands/steps outlined in the tutorial.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

ahcorde commented on citadel/spawn_urdf.md:
```diff
@@ -0,0 +1,56 @@
+## Obtaining a URDF file
+
+This tutorial assumes that the reader already has a URDF file that they'd like to load into Ignition Gazebo.
+If you are looking for a URDF file to use, you can use the `rrbot` robot model in the [`gazebo_ros_demos`](https://github.com/ros-simulation/gazebo_ros_demos) package, or use one of the examples listed at https://wiki.ros.org/urdf/Examples.
```
```suggestion
If you are looking for a URDF file to use, you can use the `rrbot` robot model in the [`gazebo_ros_demos`](https://github.com/ros-simulation/gazebo_ros_demos) package, or use one of the examples listed at [https://wiki.ros.org/urdf/Examples](https://wiki.ros.org/urdf/Examples).
```

ahcorde commented on citadel/spawn_urdf.md:
````diff
@@ -0,0 +1,56 @@
+To spawn a URDF model in Ignition Gazebo, we will start a world and make use of the world's `create` service, which uses the [EntityFactory](https://ignitionrobotics.org/api/msgs/8.1/classignition_1_1msgs_1_1EntityFactory.html) message type.
+
+Start by launching an empty world in Ignition Gazebo:
+```
````
```suggestion
```bash
```
Same in below

ahcorde reviewed (approved):
two small comments, but LGTM

jennuine commented on citadel/spawn_urdf.md:
```diff
@@ -0,0 +1,56 @@
+This tutorial will cover how to spawn a URDF model in Ignition Gazebo.
+URDF files are often used in ROS to represent robot models.
+While SDF can describe a world with multiple robot models, URDF can only describe one robot model.
+More information about URDF can be found at https://wiki.ros.org/urdf.
```
```suggestion
More information about URDF can be found at [https://wiki.ros.org/urdf](https://wiki.ros.org/urdf).
```

jennuine commented on citadel/spawn_urdf.md:
````diff
@@ -0,0 +1,56 @@
+
+The following command spawns the URDF file `model.urdf` into the Ignition Gazebo world as a model named `urdf_model`:
+```
+ign service -s /world/empty/create --reqtype ignition.msgs.EntityFactory --reptype ignition.msgs.Boolean --timeout 1000 --req 'sdf_filename: "model.urdf", name: "urdf_model"'
````
```suggestion
ign service -s /world/empty/create --reqtype ignition.msgs.EntityFactory --reptype ignition.msgs.Boolean --timeout 1000 --req 'sdf_filename: "/path/to/model.urdf", name: "urdf_model"'
```

scpeters replied:
as an aside, I wish `ign service` could autocomplete like the `ros2 service` command

jennuine reviewed (approved):
Looks good and worked for me. Left some minor suggestions

scpeters commented on citadel/spawn_urdf.md:
```diff
@@ -0,0 +1,56 @@
+
+Now that we have found the service that has an `EntityFactory` request type, we can call this service and pass the URDF file to the service's request so that the robot represented by this URDF file is spawned in the Ignition Gazebo world.
+We do this by setting the desired URDF file to the `sdf_file_name` field of the `EntityFactory` request message.
+[SDFormat](https://ignitionrobotics.org/libs/sdformat) will then internally convert the URDF file to an SDF representation, and load this into the running world.
```
we generally use `SDFormat` to refer to the specification and `libsdformat` to refer to the parser library. So perhaps: "The [libsdformat](...) library will then internally convert..."

adlarkin commented:
I have addressed all of the review feedback. This tutorial should be available in all supported Ignition releases though - should I copy/paste the files I've added to the `citadel` directory to the `dome`, `edifice` and `fortress` directories? Or is there a better approach that doesn't require copy/pasting the files?

jennuine commented:
> should I copy/paste the files I've added to the citadel directory to the dome, edifice and fortress directories? 

Per sim weekly VC, yes, the tutorials should be copied to those directories

adlarkin commented:
The tutorial has been added to dome, edifice and fortress in https://github.com/ignitionrobotics/docs/pull/219/commits/eb50ef3ebe83be09437c5f31823b03f8816c98c1. I'll go ahead and merge this.
